test/libsolidity/util/TestFileParser.cpp

```diff
@@ -503,9 +503,9 @@ void TestFileParser::Scanner::scanNextToken()
 		return TokenDesc{Token::Identifier, _literal};
 	};
 
-	auto selectToken = [this](Token _token, std::string const& _literal = "") -> TokenDesc {
+	auto selectToken = [this](Token _token, std::optional<std::string> _literal = std::nullopt) -> TokenDesc {
 		advance();
-		return make_pair(_token, !_literal.empty() ? _literal : formatToken(_token));
+		return make_pair(_token, _literal.has_value() ? std::move(*_literal) : formatToken(_token));
```

Comment:
Can data actually be moved out of `optional`?

Reply:
It can be moved, based on [this](https://gist.github.com/hrkrshnn/07834c45cc8c270205fab2fc688c0f6f) test.